workbench/tests/renderer/invest_subprocess.test.js

```diff
@@ -0,0 +1,405 @@
+  const fakeWorkspace = 'foo_dir';
+  const logfilePath = path.join(fakeWorkspace, 'invest-log.txt');
+  // invest always emits this message, and the workbench always listens for it:
+  const stdOutLogfileSignal = `Writing log messages to [${logfilePath}]`;
+  const stdOutText = 'hello from invest';
+  const investExe = 'foo';
+
+  // We need to mock child_process.spawn to return a mock invest process,
+  // And we also need an interface to that mocked process in each test
+  // so we can push to stdout, stderr, etc, just like invest would.
+  function getMockedInvestProcess() {
+    const mockInvestProc = new events.EventEmitter();
+    mockInvestProc.pid = -9999999; // not a plausible pid
+    mockInvestProc.stdout = new Stream.Readable({
+      read: () => {},
+    });
+    mockInvestProc.stderr = new Stream.Readable({
+      read: () => {},
+    });
+    if (process.platform !== 'win32') {
+      jest.spyOn(process, 'kill')
+        .mockImplementation(() => {
+          mockInvestProc.emit('exit', null);
+        });
+    } else {
+      exec.mockImplementation(() => {
+        mockInvestProc.emit('exit', null);
+      });
+    }
+    spawn.mockImplementation(() => mockInvestProc);
+    return mockInvestProc;
+  }
+
+  beforeAll(() => {
+    setupInvestRunHandlers(investExe);
+    setupInvestLogReaderHandler();
+    // mock request/response for invest usage-logging
+    const response = {
+      ok: true,
+      text: async () => 'foo',
+    };
+    fetch.mockResolvedValue(response);
+  });
+
+  afterAll(() => {
+    removeIpcMainListeners();
+  });
+
+  beforeEach(() => {
+    getSpec.mockResolvedValue(spec);
+    fetchValidation.mockResolvedValue([]);
+    getInvestModelNames.mockResolvedValue(
+      { Carbon: { model_name: modelName } }
+    );
+    uiConfig.UI_SPEC = mockUISpec(spec, modelName);
+    // mock the request to write the datastack file. Actually write it
+    // because the app will clean it up when invest exits.
+    writeInvestParameters.mockImplementation((payload) => {
+      fs.writeFileSync(payload.filepath, 'foo');
+      return Promise.resolve({ text: () => 'foo' });
+    });
+  });
+
+  afterEach(async () => {
+    await InvestJob.clearStore();
+    uiConfig.UI_SPEC = UI_SPEC;
+  });
+
+  test('exit without error - expect log display', async () => {
+    const mockInvestProc = getMockedInvestProcess();
+    const {
+      findByText,
+      findByLabelText,
+      findByRole,
+      getByRole,
+      queryByText,
+    } = render(<App />);
+
+    const carbon = await findByRole(
+      'button', { name: MOCK_MODEL_TITLE }
+    );
+    await userEvent.click(carbon);
+    const workspaceInput = await findByLabelText(
+      `${spec.args.workspace_dir.name}`
+    );
+    await userEvent.type(workspaceInput, fakeWorkspace);
+    const execute = await findByRole('button', { name: /Run/ });
+    await userEvent.click(execute);
+    await waitFor(() => expect(execute).toBeDisabled());
+
+    act(() => {
```

Comment:
rare instance of needing to use `act` because the actions we're wrapping here are not managed by `react-testing-library`